jdaviz/configs/cubeviz/plugins/viewers.py

```diff
@@ -92,17 +93,16 @@ def on_mouse_or_key_event(self, data):
                 self.label_mouseover.reset_coords_display()
 
             # Extract data values at this position.
-            # Check if shape is [x, y, z] or [x, y] and show value accordingly.
-            if (-0.5 < x < image.shape[0] - 0.5 and -0.5 < y < image.shape[1] - 0.5
-                    and hasattr(visible_layers[0], 'attribute')):
-                attribute = visible_layers[0].attribute
-                if len(image.shape) == 3:
-                    value = image.get_data(attribute)[int(round(x)), int(round(y)),
-                                                      self.state.slices[-1]]
-                elif len(image.shape) == 2:
-                    value = image.get_data(attribute)[int(round(x)), int(round(y))]
-
-                unit = image.get_component(attribute).units
+            # Check if shape is [x, y, z] or [y, x] and show value accordingly.
+            attribute = active_layer.attribute
+            unit = image.get_component(attribute).units
+            if ((len(image.shape) == 3) and (-0.5 < x < (image.shape[0] - 0.5))
+                    and (-0.5 < y < (image.shape[1] - 0.5)) and hasattr(visible_layers[0], 'attribute')):
+                value = image.get_data(attribute)[int(round(x)), int(round(y)), self.state.slices[-1]]
+                self.label_mouseover.value = f'{value:+10.5e} {unit}'
+            elif ((len(image.shape) == 2) and (-0.5 < x < (image.shape[1] - 0.5))
```

Comment:
I had to swap X/Y indices depending on whether it is a cube or not. Why is it so confusing? Is this supposed to happen or a symptom of something else?